aderyn_core/src/detect/experimental/demo_parent_chain.rs

```diff
@@ -0,0 +1,126 @@
+use std::{collections::BTreeMap, error::Error};
+
+use crate::{
+    ast::{self, NodeID},
+    capture,
+    context::{
+        browser::GetParentChain,
+        workspace_context::{ASTNode, WorkspaceContext},
+    },
+    detect::detector::{IssueDetector, IssueDetectorNamePool, IssueSeverity},
+};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct ParentChainDemonstrator {
+    // Keys are source file name and line number
+    found_instances: BTreeMap<(String, usize), NodeID>,
+}
+
+/*
+
+In ParentChainContract.sol, there is only 1 assignment done. The goal is to capture it first.
+Then walk up it's parent tree and capture a) contract definition b) if statement c) for statement
+We omit the function definition for the sake of example
+
+*/
+
+impl IssueDetector for ParentChainDemonstrator {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        for assignment in context.assignments.keys() {
+            // Retrieve Parent Chain (from closest to farthest)
+            let parents = GetParentChain::of(assignment, context);
+            {
+                println!("Parent Chain (from closest to farthest)\n---------");
```

Comment:
I'm trying to think of what I would prefer:
1. This functionality (chain in a vector)
2. `.parent(),` with the ability to do. `parent().parent().parent()` if I want to go three levels up

To be honest, I think I prefer version 2, but I'm not sure yet @TilakMaddy

Reply:
@alexroan There's a PR for that already....

https://github.com/Cyfrin/aderyn/pull/168 !!